google-cloud-storage/src/main/java/com/google/cloud/storage/GrpcStorageImpl.java

```diff
@@ -932,7 +932,40 @@ public Acl getDefaultAcl(String bucket, Entity entity) {
 
   @Override
   public boolean deleteDefaultAcl(String bucket, Entity entity) {
-    return throwNotYetImplemented(fmtMethodName("deleteDefaultAcl", String.class, Entity.class));
+    try {
+      com.google.storage.v2.Bucket resp = getBucketDefaultAcls(bucket);
+      String encode = codecs.entity().encode(entity);
+
+      Predicate<ObjectAccessControl> entityPredicate = objectAclEntityOrAltEq(encode);
+
+      List<ObjectAccessControl> currentDefaultAcls = resp.getDefaultObjectAclList();
+      ImmutableList<ObjectAccessControl> newDefaultAcls =
+          currentDefaultAcls.stream()
+              .filter(entityPredicate.negate())
+              .collect(ImmutableList.toImmutableList());
+      if (newDefaultAcls.equals(currentDefaultAcls)) {
+        // we didn't actually filter anything out, no need to send an RPC, simply return false
+        return false;
```

Comment:
What should we add to the javadocs?

This is sort of specific to the fact that we're emulating what was previously done via a single blind RPC.

Reply:
Looks like I glossed over the need to make a request to get Acl List first; Why didn't you call the RPC to delete Acl instead of querying existing bucketdefaultAcl first?

Reply:
We are patching the full default acl list of the bucket. 
1. We get the full list
2. filter out any existing acl for the specified entity
3. If there is no change, quickly return (this line)
4. Otherwise, create an update bucket request
5. Send update bucket request
6. return true if the acl is no longer in the buckets default acl from the response